mathics/builtin/forms.py

```diff
@@ -34,224 +40,47 @@
-    summary_text = "format as a table"
-
-    def apply_makeboxes(self, table, f, evaluation, options):
-        """MakeBoxes[%(name)s[table_, OptionsPattern[%(name)s]],
-        f:StandardForm|TraditionalForm|OutputForm]"""
-
-        dims = len(get_dimensions(table, head=SymbolList))
-        depth = self.get_option(options, "TableDepth", evaluation)
-        depth = expr_min((Integer(dims), depth))
-        depth = depth.value
-        if depth is None:
-            evaluation.message(self.get_name(), "int")
-            return
-
-        if depth <= 0:
-            return Expression(SymbolMakeBoxes, table, f)
-        elif depth == 1:
-            return GridBox(
-                ListExpression(
-                    *(
-                        ListExpression(Expression(SymbolMakeBoxes, item, f))
-                        for item in table.elements
-                    ),
-                )
-            )
-            # return Expression(
-            #    'GridBox', Expression('List', *(
-            #        Expression('List', Expression('MakeBoxes', item, f))
-            #        for item in table.elements)))
-        else:
-            new_depth = Expression(SymbolRule, SymbolTableDepth, Integer(depth - 2))
-
-            def transform_item(item):
-                if depth > 2:
-                    return Expression(Symbol(self.get_name()), item, new_depth)
-                else:
-                    return item
-
-            return GridBox(
-                ListExpression(
-                    *(
-                        ListExpression(
-                            *(
-                                Expression(SymbolMakeBoxes, transform_item(item), f)
-                                for item in row.elements
-                            ),
-                        )
-                        for row in table.elements
-                    ),
-                )
-            )
-
-
-class MatrixForm(TableForm):
-    """
-    <dl>
-      <dt>'MatrixForm[$m$]'
-      <dd>displays a matrix $m$, hiding the underlying list
-        structure.
-    </dl>
-
-    >> Array[a,{4,3}]//MatrixForm
-     = a[1, 1]   a[1, 2]   a[1, 3]
-     .
-     . a[2, 1]   a[2, 2]   a[2, 3]
-     .
-     . a[3, 1]   a[3, 2]   a[3, 3]
-     .
-     . a[4, 1]   a[4, 2]   a[4, 3]
-
-    ## Issue #182
-    #> {{2*a, 0},{0,0}}//MatrixForm
-     = 2 a   0
-     .
-     . 0     0
-    """
-
-    summary_text = "format as a matrix"
-
-    def apply_makeboxes_matrix(self, table, f, evaluation, options):
-        """MakeBoxes[%(name)s[table_, OptionsPattern[%(name)s]],
-        f:StandardForm|TraditionalForm]"""
-
-        result = super(MatrixForm, self).apply_makeboxes(table, f, evaluation, options)
-        if result.get_head_name() == "System`GridBox":
-            return Expression(
-                SymbolRowBox, ListExpression(String("("), result, String(")"))
-            )
-        return result
-
-
-class StringForm(Builtin):
+class _FormBaseClass(Builtin):
```

Comment:
This should not be private. Users writing a Form extension should be able to use this without guilt, same as they would for `Builtin` or `Predefined`.

Reply:
OK, but then this should be moved to `mathics.builtin.base`

Reply:
It doesn't feel like that to me. This may grow with all sorts of Form specific methods and that would be better put closer to its use, here. 

I have seen this kind of thing say with symbols. I am not sure Format symbols should be in symbols. What happens is that we get these modules that are large and require or touch stuff that isn't that related to them. 

I had considered adding a new directory here, forms, and then adding this as `base` inside that. And also the non `$OutputForms` like `StringForm`, `HornerForm`, `ScientificForm` would go there, and the things in `$OutputForm` would be in another module too. 

But right now I thought to tread lightly and not cause too much disruption since this doesn't fully work yet.

Reply:
OK, I think we can do it using #584